### DISCUSSION TITLE
Beta releases

### DISCUSSION THREAD

**Discussion by SiAdcock** [guardian/csnx#438] [Team Posts]
I've found it useful to publish pre-releases of new minor and major versions of Source. This allows me to integration test the release with dotcom-rendering or my [dedicated smoke test repo](https://github.com/guardian/source-shakedown/) to ensure I'm not breaking anything. I expect it may also help teams who are waiting on a feature that hasn't been published in a stable version.

[NPM](https://docs.npmjs.com/cli/publish) and [Yarn](https://classic.yarnpkg.com/en/docs/cli/tag/) both allow us to tag releases. By default releases are tagged as `stable`. When installing a dependency, if no tag is specified, the most recently published stable release will be installed.

NPM and Yarn are unopinionated on tag names for pre-releases. They both offer suggestions including `beta` and `canary`. 

## Proposal

I think it would be useful to standardise on `beta` as the tag name for pre-releases, so consumers could do the following without thinking:

```sh
$ yarn add @guardian/src-foundations@beta
```

```sh
$ npm install @guardian/src-foundations@beta
```

**Comment by JamieB-gu** [comment]
This looks useful!

How do you go about specifying a version along with the beta tag? So, say the latest is v2.3, can you still access v2.2 beta?

**Comment by SiAdcock** [comment]
I think the beta tag only gives you the most recently published version that was tagged with beta. You could still access the v2.2 beta using the semver version, if known e.g. 

```sh
& npm install @guardian/src-foundations@2.2.0-rc.0
```

We should also standardise how we do pre-releases using semver as there is [no One Way of denoting a pre-release](https://semver.org/#spec-item-9).
